cfme/tests/infrastructure/test_vmware_provider.py

```diff
@@ -375,13 +376,97 @@ def test_esxi_reboot_not_orphan_vms(appliance, provider):
                 vm.update_attributes(:uid_ems => SecureRandom.uuid);\
                 puts \"VM_ID=#{vm.id} name=[#{vm.name}] uid=#{vm.uid_ems}\"'"
     result = appliance.ssh_client.run_rails_command(command)
-    logger.info("Output of Rails command was {}".format(result.output))
     provider.refresh_provider_relationships()
     assert result.success, "SSH Command result was unsuccessful: {}".format(result)
     if not result.output:
+        logger.info("Output of Rails command was {}".format(result.output))
         vm_name = re.findall(r"\[.+\]", result.output)[0].split('[')[1].split(']')[0]
         vm = appliance.collections.infra_vms.instantiate(name=vm_name, provider=provider)
         view = vm.load_details(from_any_provider=True)
         power_state = view.entities.summary('Power Management').get_text_of('Power State')
         assert not power_state == 'orphaned'
         assert not power_state == 'archived'
+
+
+@pytest.mark.tier(1)
+@pytest.mark.provider([VMwareProvider], selector=ONE_PER_TYPE, override=True)
+@pytest.mark.meta(automates=[1688900])
+def test_switches_class_present_ems(appliance, provider):
+    """
+    Under this customer requested enhancement, ems should have switches class.
+
+    Bugzilla:
+        1688900
+
+    Polarion:
+        assignee: kkulkarn
+        casecomponent: Infra
+        caseimportance: critical
+        initialEstimate: 1/2h
+        testtype: integration
+        testSteps:
+            1.Add VMware provider to CFME
+            2.SSH to CFME appliance and perform following steps in rails console
+                '''
+                $evm = MiqAeMethodService::MiqAeService.new(MiqAeEngine::MiqAeWorkspaceRuntime.new)
+                p = $evm.vmdb(:ems_infra).first
+                p.class.name
+                p.switches.first.class.name
+                '''
+        expectedResults:
+            1.Provider added successfully and is refreshed
+            2.p.switches.first.class.name returns exit status 0(success) and lists class name
+            containing HostVirtualSwitch
+    """
+    command = """'$evm = MiqAeMethodService::MiqAeService.new(MiqAeEngine::MiqAeWorkspaceRuntime.new);\
+                p = $evm.vmdb(:ems_infra).first;\
+                p.class.name;\
+                puts "class name [#{p.switches.first.class.name}]"'
+                """
+    result = appliance.ssh_client.run_rails_command(command)
+    assert result.success, "SSH Command result was unsuccessful: {}".format(result)
+    logger.info("Output of Rails command was {}".format(result.output))
+    assert result.output, "Output is blank for ssh command"
```

Comment:
optional: No need of this asset as we are asserting `.success` and `HostVirtualSwitch` in `.output`

Reply:
Agree here, the assert below is taking care of this